### PR TITLE
Allow number of executor cores to have fractional values

### PR DESCRIPTION
**Pull request by liyinan926**
This commit tries to solve issue #359 by allowing the `spark.executor.cores` configuration key to take fractional values, e.g., 0.5 or 1.5. The value is used to specify the cpu request when creating the executor pods, which is allowed to be fractional by Kubernetes. When the value is passed to the executor process through the environment variable `SPARK_EXECUTOR_CORES`, the value is rounded up to the closest integer as expected by the `CoarseGrainedExecutorBackend`. I expect most use cases of this are to use a fractional of 1 CPU core, e.g., 0.1 or 0.5.

Signed-off-by: Yinan Li <ynli@google.com>

### CODE REVIEW

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala [hunk 1]
```diff
@@ -388,7 +388,8 @@ private[spark] class KubernetesClusterSchedulerBackend(
     val requiredEnv = Seq(
       (ENV_EXECUTOR_PORT, executorPort.toString),
       (ENV_DRIVER_URL, driverUrl),
-      (ENV_EXECUTOR_CORES, executorCores),
+      // Executor backend expects integral value for executor cores, so round it up to an int.
+      (ENV_EXECUTOR_CORES, math.ceil(executorCores.toDouble).toInt.toString),
```
Comment (foxish) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala]: We could make it a double when we fetch it in [KubernetesClusterSchedulerBackend#L110]( https://github.com/apache-spark-on-k8s/spark/blob/8b3248fcca89e9958a65c7519df18288c097c74e/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala#L110). That can be replaced by `conf.getDouble(..., 1.0)`

**Comment by foxish** [comment]
LGTM except for that one comment. 
Thanks @liyinan926

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala (hunk 1 above)
Comment (liyinan926) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala]: Done.

**Comment by liyinan926** [comment]
Thanks @foxish for reviewing!

**Pull request merged**